pkg/lib/verifier/rules/pcr_matches_constant.go

```diff
@@ -9,55 +9,94 @@ package rules
 //
 
 import (
-	"github.com/intel-secl/intel-secl/v3/pkg/hvs/constants/verifier-rules-and-faults"
+	"reflect"
+
+	constants "github.com/intel-secl/intel-secl/v3/pkg/hvs/constants/verifier-rules-and-faults"
 	"github.com/intel-secl/intel-secl/v3/pkg/lib/flavor/common"
 	"github.com/intel-secl/intel-secl/v3/pkg/lib/host-connector/types"
 	"github.com/intel-secl/intel-secl/v3/pkg/model/hvs"
 	"github.com/pkg/errors"
 )
 
-func NewPcrMatchesConstant(expectedPcr *types.Pcr, marker common.FlavorPart) (Rule, error) {
-	if expectedPcr == nil {
-		return nil, errors.New("The expected PCR cannot be nil")
-	}
+//NewPcrMatchesConstant collects the expected PCR values
+func NewPcrMatchesConstant(expectedPcr *types.Pcr, expectedPcrLog *types.PCRS, marker common.FlavorPart) (Rule, error) {
```

Comment:
PcrMatchesConstants does not have anything to do with event-logs (what is the expectedPcrLog parameter here for?).

Reply:
expectedPcr parameter is for exsi flow and expectedPcrLog parameter is for linux flow.
This expectedPcrLog parameter has not only event log data but also the pcr information which is of the below struct type:

type PCRS struct {
PCR PCR json:"pcr" //required
Measurement string json:"measurement" //required
PCRMatches bool json:"pcr_matches,omitempty"
EventlogEqual *EventLogEqual json:"eventlog_equals,omitempty"
EventlogIncludes []EventLogCriteria json:"eventlog_includes,omitempty"
}

**Note(rule_factory.go) :** This expectedPcrLog data was taken from the flavor part by parsing the rule field name. (in the case if it has "PCRMatches == true").


And, we are using these two expected pcr entries as now both the linux and exsi flow are having different structures (old and new implementation). When we start implementing dynamic verification for exsi flow, we can keep it as a single expected pcr entry as these flows will have the same structure.